src/tlo/methods/wasting.py

```diff
@@ -1,41 +1,1298 @@
+                                   & (df.un_clinical_acute_malnutrition == 'SAM')]
+        for person in children_with_sam.index:
+            # update clinical symptoms for severe wasting
+            self.wasting_clinical_symptoms(person_id=person)
 
     def initialise_simulation(self, sim):
+        """Prepares for simulation:
+        * Schedules the main polling event
+        * Schedules the main logging event
+        """
+
+        # schedule wasting pool event
+        sim.schedule_event(WastingPollingEvent(self), sim.date + DateOffset(months=3))
+
+        # schedule wasting logging event
+        sim.schedule_event(WastingLoggingEvent(self), sim.date + DateOffset(months=12))
+
+    def on_birth(self, mother_id, child_id):
+        """Initialise properties for a newborn individual.
+        :param mother_id: the mother for this child
+        :param child_id: the new child
+        """
+
+        df = self.sim.population.props
+
+        # Set initial properties
+        df.at[child_id, 'un_ever_wasted'] = False
+        df.at[child_id, 'un_WHZ_category'] = 'WHZ>=-2'  # not undernourished
+        df.at[child_id, 'un_clinical_acute_malnutrition'] = 'well'
+        df.at[child_id, 'un_last_wasting_date_of_onset'] = pd.NaT
+        df.at[child_id, 'un_acute_malnutrition_tx_start_date'] = pd.NaT
+        df.at[child_id, 'un_sam_death_date'] = pd.NaT
+        df.at[child_id, 'un_am_bilateral_oedema'] = False
+        df.at[child_id, 'un_am_MUAC_category'] = '>=125mm'
+        df.at[child_id, 'un_am_treatment_type'] = 'not_applicable'
+
+    def get_probs_or_odds_wasting(self, agegp: str, get_odds: bool = False) -> Union[float, int]:
+        """
+        This function will calculate the WHZ scores by categories and return probability or odds of severe wasting
+        for those with wasting status
+        :param agegp: age grouped in months
+        :param get_odds: when set to True, this argument will cause this method return the odds of severe wasting to be
+        used for scaling wasting prevalence linear model
+        :return:
+        """
+        # generate random numbers from N(meean, sd)
+        mean, stdev = self.parameters[f'prev_WHZ_distribution_age_{agegp}']
+        whz_normal_distribution = norm(loc=mean, scale=stdev)
+
+        # get all wasting: WHZ <-2
+        probability_less_than_minus2sd = 1 - whz_normal_distribution.sf(-2)
+
+        if get_odds:
+            # convert probability to odds and return the odds
+            return probability_less_than_minus2sd / (1 - probability_less_than_minus2sd)
+
+        # get severe wasting zcores: WHZ <-3
+        probability_less_than_minus3sd = 1 - whz_normal_distribution.sf(-3)
+
+        # make WHZ <-2 as the 100% and get the adjusted probability of severe wasting within overall wasting
+        proportion_severe_in_overall_wasting = probability_less_than_minus3sd * probability_less_than_minus2sd
```

Comment:
This should be a proportion, not a product a product (as suggested).
```suggestion
        proportion_severe_in_overall_wasting = probability_less_than_minus3sd / probability_less_than_minus2sd
```